chunk/chunk_store_test.go

```diff
@@ -75,7 +79,8 @@ func TestChunkStore(t *testing.T) {
 			"bar":  "beep",
 			"toms": "code",
 		},
-		Data: []byte{},
+		Encoding: chunk.DoubleDelta,
+		Data:     chunks[0],
```

Comment:
I'd be a lot happier if there were direct tests that showed that `reader` and `decode` were the reverse of each other. Also if there were tests that exercised the `metadataInIndex` path.